M2/Macaulay2/m2/last.m2

```diff
@@ -37,12 +37,21 @@ addStartFunction(
      	       dismiss "User";
 	       newPackage("User",
 		   Headline       => "default package for interpreter interactions",
-		   DebuggingMode  => true,
-		   PackageImports => if isMember("--no-preload", commandLine) then {} else Core#"preloaded packages");
+		   DebuggingMode  => true);
 	       path = prepend("./",path); -- now we search also the user's current directory, since our files have already been loaded
 	       path = unique apply( path, minimizeFilename);	    -- beautify
 	       allowLocalCreation User#"private dictionary";
-	       );
+	       )))
+
+addStartFunction( () -> (
+	if not noinitfile
+	then load(applicationDirectory() | "init.m2")))
+
+addStartFunction( () -> (
+	if not isMember("--no-preload", commandLine)
+	then for pkg in Core#"preloaded packages" do needsPackage pkg))
+
+addStartFunction( () -> (
```

Comment:
what's the advantage of having several start functions here instead of one?

Reply:
More generally, I've considered whether we should move these start functions to the end of `Core` instead. This file seems very chaotic.

Reply:
I figured more granularity -- one function that does one thing -- might be good.  I don't have a strong opinion, though.

Reply:
Move all of them to Core.m2, or just the ones in last.m2?  There a few in other files that kind of make sense where they are (like overwriting `seenWarnings` in programs.m2).